\chapter{Modelling Autofluorescence in Skin for Novel Biomarkers of Cardiovascular Diseases}
\label{chap:salvo}

% explain it is hard to determine which fluros contribute to signal, and where from within skin can try to sue mcrt to alleviate this

% motivation
% explain fluro method
% explain skin model
% explain nelder mead
% explain filter choice

% results in "2d" 
% results in 3d + higher
% future work

\section{Introduction}


\Gls*{cvds} are the leading cause of death in the world~\cite{whodeath}.
It is estimated that around 18 million people died in 2016 from \gls*{cvds}, accounting for 31\% of global deaths~\cite{whodeath}.
Despite decreasing burden of \gls*{cvds} in the UK, it was still the number two cause of death in the UK in 2014~\cite{bhatnagar2016trends}.

Currently risk factors are used to try to determine if a patient has \gls*{cvd}.
However, these risk factors are only a ``causal pathway leading to the disease''~\cite{vasan2006biomarkers}.

The risk factors such as high blood pressure, smoking, diabetes, physical inactivity, and dyslipidemia do not fully explain incidence of disease~\cite{olsen2010assessment,folsom2013classical}.
Therefore research has moved to examining more novel biomarkers for detecting the disease.
Among these novel biomarkers, the autofluorescence response of tissue is of much interest.

Autofluorescence is the natural fluorescence released by biological structures upon excitation by light.
Autofluorescence is particularly attractive as a biomarker as it requires no exogenous dyes, which can be toxic, non-specific, or interfere with biological function~\cite{kollias1998endogenous}.
In tissue there are several fluorophores responsible for this autofluorescent response, including: NADH (nicotinamide adenine dinucleotide), structural proteins like collagen and elastin, aromatic amino acids (tyrosine and tryptophan), porphyrins, and FAD (flavin adenine nucleotide)~\cite{monici2005cell}.
Changes in the autofluorescent response of tissue has been linked to cancer, Alzheimers, diabetes and \gls*{cvds}~\cite{drakaki2009laser,pu2013native,ramanujam2000fluorescence,tarnawska2018pilot,van2019skin}.
Autofluorescence can be linked to these diseases as the fluorophores responsible for the autofluorescence either originate in the mitochondria, or are involved in important biochemical pathways that regulate apoptosis, free-radical generation, oxidative stress and biomolecular sensing of glucose, oxygen and nitric oxide.

NADH has recently been the subject of interest in particular as a biomarker for \gls*{cvds}~\cite{akbar2014vivo,elahi2009oxidative,blacker2016investigating}.
NADH is a intracellular co-enzyme that is a biomarker for metabolic activity, and mitochondrial anomalies. 
It is involved in mitochondrial function, energy metabolism, calcium homoeostasis, gene expression, oxidative stress, ageing, apoptosis and glycolysis (see~\cref{fig:nadhfadpath}).
Therefore if there is dysfunction, then the autofluorescent signal from NADH will be affected, thus it can be used as biomarker for disease.

Despite their appeal using autofluorescence to diagnose and assess disease risk more research needs to be carried out on various unknowns: information on the location of fluorescence, which fluorophores contribute to the signal and how much, how much the optics of tissue affect the signal, and the variability of the signal over different locations on the body.
This chapter aims to determine how much the tissue optics affects the autofluorescent signal, which fluorophores contribute to the signal, and from which layer of the skin do they contribute from.
Finally we introduce our ameboaMCRT algorithm, created to determine the relative concentrations of the intrinsic fluorophores in the skin to assess clinical outcomes.


\begin{figure}[!htpb]
  \centering
  \includegraphics[width=0.75\textwidth]{nadh-cycle.png}
  \caption{Simplified schematic showing the roles of NADH and FAD in various different metabolic pathways. The star boxes indicate fluorescing forms of NADH and FAD. Taken from Croce \textit{et al.}~\cite{croce2014autofluorescence}.}
  \label{fig:nadhfadpath}
\end{figure}


\FloatBarrier

\section{Skin Model}

So far in this thesis all tissue models have been simplified, by assuming that tissue is a homogeneous structure with uniform optical properties.
However, this is not the case in reality.
Tissue is un-homogeneous with nonuniform optical properties.
However, to create a one to one model of tissue in a simulation is impractical due to the resolution required to resolve all the constituent parts of the tissue down to the cell level.
Therefore, we need to make a compromise between reality and what is possible to model efficiently.
This section presents a five layer model of human skin. 

Dermatologists split the skin into several layers based upon the morphology, function, and contents of each layer~\cite{freedberg1999fitzpatrick,zaidi2010dermatology}.
The layers are named from outer layer to inner most layer: stratum corenum, stratum lucidum, stratum granulosum, stratum spinosum, stratum basale, papillary dermis, reticular dermis, and hypodermis.
As not all these layers are optically distinct, are too small to model or not present in a given location on the body.
Therefore, we simplify this into just 5 layers: stratum corneum, epidermis, papillary dermis, reticular dermis, and hypodermis, with the epidermis compromising of stratum lucidum, stratum granulosum, stratum spinosum, stratum basale.~\Cref{fig:skinexample} shows the geometry of this model.

\begin{figure}[!htpb]
    \centering
    \includegraphics[width=0.7\textwidth]{Skin_layers.pdf}
    \caption{Illustration of our five layer skin model.}
    \label{fig:skinexample}
\end{figure}

Each of these layers have various amounts of chromophores and scatterers.
To accurately model these various chromophores and scatterers, and therefore the skin, we must discuss the biological make up and spatial structure of the skin.

\subsubsection*{Stratum Corenum} % (fold)
\label{sub:stratum}

The outer most layer of the skin is the stratum corenum.
This layer mostly consists of dead skin cells (keratinocytes).
Keratinocytes, which make up approximately 80 percent of the cells in the epidermis, are born in the stratum basale and live for approximately 14 days.
During this period they move upwards to the surface of the skin undergoing a series of complex morphological and metabolic events, which ends with the cells undergoing apoptosis by the time the reach the stratum corenum.
The keratinocytes are flat in shape, cornified and stacked on top each other in this layer.
The function of this layer is to be a protection barrier to prevent damage, infection and diffusion of unwanted chemicals further into the skin.
The stratum corenum also prevents water loss and provides some UV protection~\cite{freedberg1999fitzpatrick,zaidi2010dermatology}.

% subsection stratum_corenum (end)

\subsubsection*{Epidermis} % (fold)
\label{ssub:epidermis}

Below the Stratum corenum is the epidermis.
The epidermis consist of several layers that are optically similar so we restrict our model to modelling this as one whole layer.
The layers that make up the epidermis are the stratum basale, stratum spinosum, stratum granulosum, and stratum lucidum\footnote{The stratum lucidium only appears in areas of the body where skin is thick, e.g the palm or the sole of the foot.}.
Each of these layers are distinct from one another, in the fact that the keratinocytes in each layer are different from one another.
In the stratum lucidium the keratinocytes are dead, and fairly flat.
The stratum granulosum the keratinocytes are grainy are becoming fairly flat in comparison to the layers below.
In the Stratum spinosum the keratinocytes appears spiny (hence the name of the layer), and are polyhedral in shape.
It is in the stratum spinosum that the cells begin to become keratinized and start the process of dying as they move upwards.
Finally the last layer of the epidermis, the startum basale, is the layer where the keratinocytes are born.
Here the keratinocytes are columnar or cuboid in shape.

The purpose of the epidermis is as before to provide a protective barrier to the underlying layers. 
In the stratum basale there is also melanocytes which produce the pigment melanin which is responsible for the color of the skin, and providing some protection from harmful UV light. 
Other types of cells found in the epidermis are Langerhans cells and Merkel cells which are part of the immune system and nervous system respectively.
Overall the epidermis provides protection from mechanical stress, flexibility to the skin, UV protection, retains water, and stops foreign bodies or chemicals from entering the skin~\cite{freedberg1999fitzpatrick,zaidi2010dermatology}.

% subsubsection epidermis (end)

\subsubsection*{Dermis} % (fold)
\label{sub:dermis}
The dermis is the layer of skin below the epidermis, and makes up the majority of the skin by size.
The cellular makeup of the dermis is different from that of the epidermis, as the dermis is mostly made up of filamentous or fibrous proteins such as elastin or collagen.
The dermis also contains various nerves and blood vessels.
The dermis is split into three layers; the papillary dermis, the reticular dermis, and the hypodermis.
The boundary between the papillary and reticular layers is demarcated by the subpapilliary plexus, which is a horizontal plane of blood vessels.
The boundary between the reticular and the hypodermis is marked by a abrupt change from fibrous to adipose tissue.
The function of the dermis is to protect from mechanical injury, thermal regulation, contains receptors of sensory stimuli, giving the skin pliability, elasticity and tensile strength~\cite{freedberg1999fitzpatrick,zaidi2010dermatology}.
 

% subsection dermis (end)

\subsection*{Optical Properties} % (fold)
\label{sub:optical_properties}


%refractive indices from mm02
%blood from louise + inglesias
%melanin from louise + layers thickness
% bilirubin from krishnaswamy04 et al
% carotene  from krishnaswamy04 et al
%water from mm02

With a discussion of what makes up the skin and what molecules contribute to the skins optical properties, this section gives an account of how our skin model models the optical properties of skin.
First we will discuss the absorbers that are found in the skin and what their absorption properties are as a function of wavelength.
Scattering properties of the several layers of tissue will then be discussed followed by a quick discussion on refractive indices and anisotropy values.
Finally a discussion on the fluorophores found in the skin will be presented.

\medskip

The first chromophore we will examine is blood.
To model blood we first split blood into its deoxygenated and oxygenated components.
This is done as the absorption coefficient differs between the two types of blood. We mix these two groups using the tissue oxygenation coefficient $S$. Blood absorption spectra are taken from S. Prahl~\cite{prahlblood}.
\Cref{eqn:oxy,eqn:blood} give the absorption coefficients for oxygenated, deoxygenated, and whole blood.


\begin{equation}
\mu_{a,Oxy/DeOxy}=150\ \ln10\ \frac{\epsilon}{64458}
\label{eqn:oxy}
\end{equation}

\begin{equation}
\mu_{a,blood}(\lambda) = S\mu_{a,Oxy}+(1-S)\mu_{a,DeOxy}
\label{eqn:blood}
\end{equation}

\noindent Where:

\indent $\epsilon$ is the extinction coefficient of hemoglobin [$cm^{-1}$];

\indent $64458$ is the molecular weight of hemoglobin [$g\ mol^{-1}$];

\indent $150$ is the normal concentration of hemoglobin in blood [$g\ L^{-1}$];

\indent and $\mu_{a,Oxy}$, $\mu_{a,DeOxy}$, and $\mu_{a,blood}$ are the absorption coefficients for oxygenated, 

\indent deoxygenated and blood respectively [$cm^{-1}$].

\medskip

We also include water in our skin model.
Water's absorption spectrum is taken from the work of Wieliczka \textit{et al} and Segelstein~\cite{wieliczka1989wedge,segelstein1981complex}.

The next chromophores are bilirubin and $\beta$-carotene.
These chromophores are yellow/orange pigments. 
Bilirubin is usually responsible for the yellow skin colour seen in people with jaundice~\cite{jacques1997developing}.
The spectra are taken from S. Prahl's compilation of PhotochemCAD data~\cite{prahlcaro,prahlbili}.
\Cref{eqn:bili,eqn:caro} give the absorption coefficients of bilirubin and $\beta$-carotene:

\begin{equation}
\mu_{a,Bilirubin}(\lambda)=\frac{\epsilon_{bilirubin}}{585}\ ln10\ C_{bilirubin}
\label{eqn:bili}
\end{equation}

\begin{equation}
\mu_{a,\beta-Caro}(\lambda)=\frac{\epsilon_{\beta-Caro}}{537}\ ln10\ C_{\beta-Caro}
\label{eqn:caro}
\end{equation}

\noindent Where:

\indent $\epsilon_{bilirubin}$ and $\epsilon_{\beta-Caro}$ are the extinction coefficients for bilirubin and $\beta$-carotene 

\indent respectively [$cm^{-1}$];

\indent $585$, and $537$ are the molecular weights of bilirubin and $\beta$-carotene [$g$];

\indent finally, $C_{bilirubin}$, and $C_{\beta-Caro}$ are the concentrations of bilirubin and $\beta$-carotene 

\indent in the skin [$g\ L^{-1}$].

\medskip

Melanin is the next chromophore we model.
To model melanin's absorption coefficient we use~\cref{eqn:mel1,eqn:melanin}, taken from~\cite{iglesias2015biophysically}.

\begin{align}
\mu_{a,eumel}(\lambda)&=6.66\times10^{11} \times \lambda^{-3.33}\label{eqn:mel1}\\
\mu_{a,phomel}(\lambda)&=2.9\times10^{15} \times \lambda^{-4.75}
\label{eqn:melanin}
\end{align}

Finally we use a base absorption coefficient to model the absorption due to the other parts of the skin that contribute to its optical properties, but individually do not have a large effect.
The equation for modelling this was taken from I. Sahdi~\cite{saidi1992transcutaneous}.

\begin{equation}
\mu_{a,base}=7.84\times10^{8}\times\lambda^{-3.255}
\label{eqn:base}
\end{equation}


\Cref{fig:absorcoeff} shows the absorption spectra for the above chromophores as a function of wavelength.

\begin{figure}[!htpb]
	\centering
	\includegraphics[width=0.75\textwidth]{absorbs.pdf}
	\caption{Absorption coefficients for the various chromophores found in our skin model.}
	\label{fig:absorcoeff}
\end{figure}

To create the five layer skin model we mix different amounts of the chromophores found in each layer, as described above.
\Cref{eqn:stratabs,eqn:epiabs,eqn:dermisabs,eqn:hypoabs} show the respective equations for the total absorption coefficient for each layer.
These equations were adapted from~\cite{meglinski2002quantitative,iglesias2015biophysically,jacques2013optical}.

\begin{equation}
\mu_{a,strat}= ((0.1 - 0.3\times10^{-4}\cdot\lambda) + (0.125(\lambda/10.))\times \mu_{a,b}(\lambda))\times(1. - W) + W\cdot\mu_{H_20}(\lambda)
\label{eqn:stratabs}
\end{equation}

\begin{equation}
\begin{split}
\mu_{a,epi}= (\nu_m \cdot (\mu_{phomel}(\lambda) + \mu_{eumel}(\lambda)) + (\mu_{a,b}(\lambda) + \ln10 \cdot \mu_{a,\beta-caro}(\lambda) \cdot C_{caro}) \\ \times (1. - \nu_m)) \times(1. - W) + W \cdot \mu_{H_2O}(\lambda)
\end{split}
\label{eqn:epiabs}
\end{equation}

\begin{equation}
\begin{split}
\mu_{a,pap/ret}=((S \cdot \mu_{a,oxy}(\lambda) + (1. - S) \cdot \mu_{a,deoxy}(\lambda) + \ln10 \cdot \mu_{a,\beta-caro}(\lambda) \cdot C_{caro} + \\ \ln10 \cdot \mu_{a,bili}(\lambda)\cdot C_{bili})\cdot B +  (1. - B)\cdot \mu_{a,b}(\lambda)) \times (1. - W) + W \cdot \mu_{H_2O}(\lambda)
\end{split}
\label{eqn:dermisabs}
\end{equation}

\begin{equation}
\begin{split}
\mu_{a,hypo}=((S\cdot \mu_{a,oxy}(\lambda) + (1. - S) \cdot \mu_{a,deoxy}(\lambda)) \times B + \mu_{a,b}(\lambda) \cdot (1. - B))\\ \times (1. - W) + W \cdot \mu_{a,H_2O}(\lambda)
\end{split}
\label{eqn:hypoabs}
\end{equation}

\noindent Where:

\indent $W$ is the volume fraction of water in a layer [-];

\indent $\nu_m$ is the volume fraction of melanin in a layer [-];

\indent $B$ is the volume fraction of blood in each layer [-];

\indent and all the other symbols have the same meanings as described above.

\medskip

\Cref{tab:optpropsvals} shows the amount of each chromophore is included in each layer, and the size adopted for each layer as well as the refractive index for each individual layer.
\Cref{fig:absoplayers} shows the total absorption coefficient as a function of wavelength for the five layers.


\begin{table}[!tbhp] 
    \begin{adjustwidth}{-.5in}{-.5in}  

  \begin{center}
  \begin{tabular}{|c|c|c|c|c|c|c|c|}
  \hline
  Layer & Thickness/ & Refractive & Blood & Melanin & Bilirubin/ & $\beta$-Carotene/ & Water\\
    &cm & index & volume/\% & volume/\% & $gL^{-1}$ & $gL^{-1}$ & volume/\%\\
  \hline
  Stratum Corenum  & 0.02 & 1.50  & 0.0 & 0.0 & 0.0  & 0.0 & 0.05\\
  Epidermis        & 0.08 & 1.34  & 0.0 & 1.0 & 0.0  & 2.1e-4 & 20.0\\
  Papillary Dermis & 0.18 & 1.40  & 6.0 & 0.0 & 0.05 & 7e-5 & 50.0\\
  Reticular Dermis & 1.82 & 1.395 & 4.5 & 0.0 & 0.05 & 7e-5 & 70.0\\
  Hypodermis       & 5.90 & 1.41  & 5.0 & 0.0 & 0.0  & 0.0 & 70.0\\

  \hline
  \end{tabular}
    \caption{Table of values used for the various concentrations and volumes fraction of the chromophores in the five layer skin model. Values taken from~\cite{krishnaswamy2004biophysically,meglinski2002quantitative,campbell20153d,iglesias2015biophysically}.}
  \label{tab:optpropsvals}
  \end{center}
      \end{adjustwidth}

\end{table}


\begin{figure}[!htpb]
  \centering
  \includegraphics[width=0.75\textwidth]{optical-props.pdf}
  \caption{Absorption coefficients for the different layers in our skin model.}
  \label{fig:absoplayers}
\end{figure}


With the absorption properties of the various chromophores in the skin defined we can now discuss the scattering properties of the skin.
As the scattering properties do not vary from layer to layer by too much we use the same equation to describe the scattering coefficient~\cite{jacques2013optical,iglesias2015biophysically,louisethesis}.

\begin{equation}
\mu'_s(\lambda)=a'\left(f_{ray}\left(\frac{\lambda}{500(nm)}\right)^4+(1-f_{ray})\left(\frac{\lambda}{500(nm)}\right)^{-b_{\text{mie}}}\right)
\label{eqn:scattrest}
\end{equation}

\begin{equation}
\mu'_s(\lambda)=1050.60\times\lambda^{-0.68}
\label{eqn:hyposcat}
\end{equation}


\noindent Where:

	$\mu'_s$ is the reduced scattering coefficient [$cm^{-1}$];

	$a'$ is a scaling factor [$cm^-1$];

	$f_{ray}$ is the fraction of Raylieigh scattering [-];

	$\lambda$ is the wavelength of light [$nm$];

	and $b_{\text{mie}}$ is the ``scattering power'' [-].

\medskip

This equation mixes both Mie and Rayleigh scattering into one equation.
The first term represents the Rayleigh scattering terms, whilst the second represents the Mie scattering term.
\Cref{fig:scatplot} shows the reduced scattering coefficients for the different layers of the skin.
\Cref{tab:valscat} show the values used in the model for the scattering equation for the different layers.

\begin{table}[!htpb]
  \centering

  \begin{tabular}{|c|c|c|c|}
  \hline

  Layer & $a'/cm^{-1}$ & $f_{ray}$ & $b_{mie}$ \\
  \hline
   Epidermis         & 66.7 & 0.29 & 0.69 \\
   Dermis  & 43.6 & 0.41 & 0.35 \\

  \hline
  \end{tabular}
  \caption{Values of the constants for the scattering equations for the different layers of our skin model. Here epidermis represents the stratum corenum and the epidermis, and dermis represents the papillary, reticular and hypodermis in our model.}
  \label{tab:valscat}

\end{table}

\begin{figure}[!htpb]
	\centering
	\includegraphics[width=0.75\textwidth]{scat-plot.pdf}
	\caption{Figure shows the reduced scattering coefficient for the different layers of our skin model.}
	\label{fig:scatplot}
\end{figure}

\FloatBarrier

Finally we discuss the anisotropy values and refractive indices of our model.

First, the anisotropy $g$ value is dependent on wavelength~\cite{louisethesis,van1989skin}.

\begin{equation}
g(\lambda)=0.62+0.29\lambda\times10^{-3}
\end{equation}

\noindent Where:

\indent g is the anisotropy value [$-$];

\indent $\lambda$ is the wavelength [$nm$].

\medskip

The refractive indices of each layer is broadly the same and does not vary form layer to layer by a large amount.
\Cref{tab:refindex} shows the refractive indices adopted for our skin model.

\begin{table}[!htpb]
  \centering

  \begin{tabular}{l|c}
  \hline
  Layer & Refractive index \\
  \hline
    Stratum corenum & 1.5 \\
    Epidermis &  1.34\\
    Papilliary dermis & 1.40 \\
    Reticular dermis &  1.395\\
    Hypodermis &  1.41\\

  \hline
  \end{tabular}
    \caption{Refractive indices used for the five layer skin model. Values are taken from~\cite{meglinski2002quantitative}.}
  \label{tab:refindex}
\end{table}

As information on wavelength dependent refractive indices is not readily available, we assume that the refractive indices are the same for all wavelength used in this chapter.

\subsection*{Fluorophores in the Skin}

The final optical property that makes up the five layer model is fluorescence.
As mentioned in the introduction there are various molecules and proteins responsible for the autofluorescent response of the skin.
These include NADH (nicotinamide adenine dinucleotide), structural proteins like collagen and elastin, aromatic amino acids (tyrosine and tryptophan), porphyrins, and FAD (flavin adenine nucleotide).
\Cref{fig:flurosshow} shows the emission and absorption spectra for these fluorophores.
As NADH and FAD are found in cells, both are typically found in all layers of the skin.
Elastin and collagen are only found in the dermis~\cite{gillies2000fluorescence}.
Tyrosine and tryptophan are found in all layers of the skin~\cite{gillies2000fluorescence}.

% stratum        nadh, fad, tyrosine, trytophan
% epidermis    nadh, fad, tyrosine, trytophan
% pap            nadh, elastin, collagen, tyrosine, trytophan
% ret            nadh, elastin, collagen, tyrosine, trytophan
% hypo           nadh, elastin, collagen, tyrosine, trytophan

% NADH ->
% FAD -> fad smae as NADH?

\begin{figure}[!htbp]
  \centering
  \includegraphics[width=.75\textwidth]{fluros.pdf}
  \caption{Top) Shows the fluorescent emission for the various different fluorophores. Bottom) Extinction coefficients for a selection fluorophores found in the skin~\cite{prahltyro,prahltryto,soltani2019deep,sun2012biomarkers,islam2013ph,evans2013magnetic,von2012fluorescence,dacosta2003molecular}.}
  \label{fig:flurosshow}
\end{figure}
% subsubsection optical_properties (end)
\FloatBarrier

\section{Modelling Fluorescence}

Fluorescence is the process in which light of a certain wavelength is incident on a molecule, the molecule absorbs the light and re-emits the light at a new longer wavelength.
This process can be illustrated by a Jablonski diagram.
A Jablonski diagram shows the electronic states of a molecule and the possible transitions between them.
This allows the illustration of various radiative and nonradiative transitions that are possible for a give  molecule, including fluorescence and phosphorescence.
\Cref{fig:Jabo} shows an example of Jablonski diagram for a molecule that absorbs light and re-emits it as fluorescence.

\begin{figure}[!htpb]
	\centering
	\includegraphics[width=0.45\textwidth]{jabo-diag.pdf}
	\caption{Jablonski diagram for PPIX. a) excitation of the ground state via absorption of a photon, b) non-radiative transition, and c) fluorescence.}
	\label{fig:Jabo}
\end{figure}

To model fluorescence from multiple fluorophores requires a change of the \gls*{mcrt} code presented thus far.
This change is to the interaction portion of the algorithm, so that it will now include the option for a packet to undergo fluorescence.
To calculate whether a packet absorbs, scatters, or fluoresces, first the probability of each of these events must be calculated.
Discussion of scattering and absorption (by the bulk medium) was described in~\cref{sec:opticalprops}.
To calculate the probability of fluorescence we first assume that the quantum yield of the molecule is unity.
This is physically unrealistic however, it does not affect the simulations accuracy as modelling a realistic quantum yield would mean that more packets would be discarded, and thus the signal to noise ratio would be worse than if we assume a quantum yield of unity.
Furthermore, information on the fluorophores, used in this work, quantum yield's is not readily available.
To calculate the probability of fluorescence the absorption coefficient of the florescent molecule must be calculated.
This is shown in~\cref{eqn:flurodef}:

\begin{equation}
\mu_f=\ln\left(10\right)\varepsilon C
\label{eqn:flurodef}
\end{equation}

Where $C$ is the concentration of the fluorophore, $\varepsilon$ is the extinction coefficient of the fluorophore, and $ln(10)$ is the natural logarithm of 10\footnote{This factor appears as historically $\varepsilon$ was measured in base 10~\cite{jacques2013optical}.}.\\

The next step is to calculate the total attenuation coefficient for a given species as in~\cref{eqn:totdef}

\begin{equation}
\mu_{t_i}=\mu_{s_i}+\mu_{a_i}+\mu_{f_i}
\label{eqn:totdef}
\end{equation}

Where as usual $\mu_a$ and $\mu_s$ are the absorption and scattering coefficients, and $\mu_f$ is the fluorescence coefficient as defined in~\cref{eqn:flurodef}.
As the absorption coefficient of fluorophores are small in comparison to the medium, and that the absorption coefficient of fluorescent molecules are generally much larger than that of their scattering coefficient, we assume that the scattering coefficient is negligible.
Finally we calculate the probability of interacting with a given species using~\cref{eqn:totspec}

\begin{equation}
P_i=\frac{\mu_{t,i}}{\sum\limits_{i=1}^{N} \mu_{t,i}}
\label{eqn:totspec}
\end{equation}

Where $P_i$ is the probability of interacting with the $i^{th}$ species, the numerator is the attenuation coefficient for $i^{th}$ species, and the denominator is the total attenuation coefficient for all the species.

\cref{algo:speciespick} shows the process used to determine which species to interact with.

\begin{center}
\begin{algorithm}[H]
\SetAlgoLined
  set $\mu_{t_{i}}$\;
  set all $P_i$'s\;
  set $\xi_1$\;
  \uIf{$\xi_1 <= P_1$}{
    set $\xi_2$\;
    \uIf{$\xi_2 <= a_m$}{
      Scatter in medium\;
     }
     \Else{
      Absorb in medium\;
     }
  }
  \uElseIf{$\xi_1 <= P_1 + P_2$}{
    Species 1 fluoresces\;
  }
  \uElseIf{$\xi_1 <= P_1 + P_2 + P_3$}{
    Species 2 fluoresces\;
  }
  \uElseIf{$\xi_1 <= P_1 + P_2 + P_3 + ... + P_n$}{
    Species n fluoresces\;
  }
  \Else{
    Error, no interaction\;
  }
\
\caption{\textit{An algorithm to determine which species to interact with. $P_1$ is the probability of interacting with the bulk medium, $P_2$ to $P_n$ is the probability of interacting with a fluorescent species, $a_m$ is the albedo of the bulk medium, $\xi_i$ is a random number, and $\mu_{tot}$ is the total attenuation coefficient of all the species summed. The error condition should never be met.}}
\label{algo:speciespick}
\end{algorithm}
\end{center}

This method allows an arbitrary number of fluorophores to be modelled.
To ensure that this method works as intended, the method is compared to experimental data taken by C.L. Campbell \textit{et al.}~\cite{louisethesis}.

C.L Campbell \textit{et al.} filled a cuvette with Intrlipid $20\%$ diluted with water, and a fluorescent agent Coproporphyrin III\@.
The total volume of this mixture was $6~ml$, consisting of $1~ml$ of Coproporphyrin III, $4.99~ml$ of water, and $0.0~1ml$ of Intrlipid $20\%$, of which $2~ml$ was pipetted into the cuvette.
$405~nm$ light was shone on a cuvette of area $10~mm \times 10~mm$, and collected via the same fiber, of diameter $600~\mu m$, and NA of 0.22.
The signal was the recorded by an \gls*{obs}.
The \gls*{obs} uses a 405~nm laser diode ($\sim1~mW$ output).
The light was delivered by the aforementioned fiber, and fluorescent light was also collected by the same fiber.


\begin{figure}[!htpb]
	\centering
	\includegraphics[width=0.75\textwidth]{cop-optprop.pdf}
	\caption{Optical properties of Coproporphyrin III. The figure on the left shows the absorption coefficient as a function of wavelength. The figure on the right shows the emission spectrum as a function of wavelength.}
	\label{fig:coporiii}
\end{figure}

The simulation is setup to mimic the experimental setup.
A medium of $10~mm^3$ is used with 1 voxel to increase the speed of computation.
As before Intralipid is assumed to be wholly scattering with no absorption, so an albedo of 1 is used.
Conversely the Coproporphyrin III is wholly absorbing with no scattering.
Coproporphyrin III absorption spectrum is as shown alongside its emission spectrum in~\cref{fig:coporiii}.
If a photon packet leaves the top face of the simulated medium, within the radius of the fiber at an angle the fiber could accept, then the packet is recorded.
The simulation is run with $10^7$ photons which yielded~\cref{fig:flurovalid}.
The algorithm presented above gave a good fit to the experimental data.

\begin{figure}[!htpb]
  \centering
  \includegraphics[width=0.75\textwidth]{Fluro-valid.pdf}
  \caption{Validation of florescence modelling technique as described above. Figure shows that the MCRT method matches closely to the experimental results.}
  \label{fig:flurovalid}
\end{figure}

\FloatBarrier

\section{Nelder-Mead Method}

To determine the contribution of individual fluorophores and their concentrations, emission profiles of these fluorophores cannot simply be fitted to autofluorescence signals.
This is due to the nonlinear effect that tissue optics have on the emission profiles.
Therefore, \gls*{mcrt} can be used to compute this effect.
This leaves the problem as an optimisation problem.
The Nelder-Mead method is an algorithm for unconstrained optimisation. 
The algorithm is based upon iteratively updating a simplex. 
A simplex is a structure in $n$-$dimensional$ space, consisting of $n+1$ points. 
Therefore in 1D, the simplex is a line, in 2D a triangle, in 3D a tetrahedron, etc. 
The Nelder-Mead method is a gradient free method, meaning that it does not require derivatives to be calculated and that the search space does not need to be smooth.
This makes it ideal for problems where derivatives are not able to be computed easily, or the search space is not smooth.
However, the NM method can also get stuck at local minima so care must be taken to avoid this.
Genetic algorithms, which do not easily get stuck in local minima, were trialled but the computational cost of using them made it infeasible. 

The NM algorithm works by removing the worst vertex of the simplex and replacing it with a ``better'' vertex calculated via a number of different operations.
These operations can be seen in~\cref{fig:NM-operations}.\\

The first step of the~\gls*{nm} method is to sort the initial vertices according to their fitness.
For $n=2$, we define $x_w$ as the ``worst'' point, $x_l$ and the ``lousy'' point, and $x_b$ the ``best'' point, such that $f(x_b)\leq f(x_l)\leq f(x_w)$, where $f(x)$ is evaluating the `fitness' of a point $x$. 
The fitness function varies from problem to problem, and usually takes the form of the function that is being optimised.

With the vertices sorted, the centroid of the simplex is calculated as in~\cref{eqn:centroid}.
The centroid is the mean of all the vertices bar the ``worst'' point.

\begin{figure}[!htbp]
    \centering
    \includegraphics[width=0.75\textwidth]{simplex-operations.pdf}
    \caption{Operations that can be preformed on a simplex for $n=2$.}
    \label{fig:NM-operations}
\end{figure}

The next step is to move the simplex via a reflection.
To calculate the new vertex via reflection~\cref{eqn:reflect} is used, where $\alpha$ is the reflection factor.
If this new point, $x_r$, is better\footnote{Here better means the point has a lower fitness score.} than the current ``best'' point then we calculate a new point in the same direction but further using the expansion operation~\cref{eqn:expand}, where $\gamma$ is the expansion factor.
If this new point, $x_e$, is better than the ``best'' point then we replace $x_w$ with $x_e$ and start the process again.
However, is $x_e$ is not better than the ``best'' point, then we discard it and replace the worst point with $x_r$ the reflected point.

If when calculating $x_r$, we find that is worse than the ``best'' point, we then check if $x_r$ is better than the `lousy' point.
If $x_r$ is better than $x_l$ then we replace the ``worst'' point and start the process again.
However, if the $x_r$ is worse than $x_l$, we then compare it to the ``worst'' point.
If $x_r$ is better than the ``worst'' point then we preform and inside contraction~\cref{eqn:insidecontract}, where $\beta$ is the contraction factor.
If this new point, $x_{ic}$, is better than the ``worst'' point then we keep it, otherwise we preform the shrink operation, shrinking the whole simplex around the ``best'' point.

If $x_r$ is not worse than the ``worst'' point then we preform an outside contraction~\cref{eqn:outsidecontract}.
This computes a new point $x_{oc}$.
If $x_{oc}$ is better than $x_w$, then we keep it, otherwise again we shrink around the ``best'' point.
The process described above is summarised in~\cref{fig:NM-algo}.

\begin{align}
c &= \frac{1}{n}\sum \limits_{i=1,i\neq w}^{n+1} x_i \label{eqn:centroid}\\
x_r &= c + \alpha(c - x_w)\label{eqn:reflect}\\
x_e &= c + \gamma(x_r - c)\label{eqn:expand}\\
x_{oc} &= c + \beta(x_r - c)\label{eqn:outsidecontract}\\
x_{ic} &= c + \beta(x_w - c)\label{eqn:insidecontract}
\end{align}

Standard values for the factors are: $\alpha=1$, $\beta=\frac{1}{2}$, and $\gamma=2$.
Though in practice these values are adjusted for the problem at hand.
For higher dimensions, i.e. where $n > 2$ F. Gao \textit{et al}. suggest that the parameters should be changed based upon how many dimensions are used for the simplex~\cite{gao2012implementing}.
The values F. Gao \textit{et al} suggest are: $\alpha=1$, $\beta=1+\tfrac{2}{n}$, $\gamma=0.75-\tfrac{1}{2n}$, and $\delta=1-\tfrac{1}{n}$.
Where $n$ is the order of dimensions.

As the Nelder-Mead method has no inbuilt convergence criteria, this must be added.
We use two different criteria based upon simplex size, and vertex fitness.
The size of the simplex is calculated using~\cref{eqn:sizeof}:

\begin{equation}
size=\sum\limits_{i=1}^{n+1}|p_{i}-p_{i+1}|
\label{eqn:sizeof}
\end{equation}

Where $p_i$ and $p_{i+1}$ are vertices in the simplex that are connected by an edge. 
If the size of the simplex falls below a pre-set value, then we preform a factorial test to see if the simplex should be restarted or if the algorithm should terminate.
The factorial test checks the space around the current simplex to ensure that we have converged to a global minima.
If the check fails then the algorithm is restarted with the current best point kept, and new vertices generated.

\begin{figure}[!htbp]
    \centering
    \includegraphics[width=0.5\textwidth]{flowchart-nelder.pdf}
    \caption{Nelder-Mead decision tree.}
    \label{fig:NM-algo}
\end{figure}

The other convergence criteria is a check to see if the best point is ``good enough''.
The current best point is compared to a pre-set fitness value.
If the best point is better than the pre-set value then the algorithm terminates.

\FloatBarrier
\section{Validation}
The~\gls*{nm} method was coded in modern Fortran, so that it could be easily interfaced with the~\gls*{mcrt} code developed as part of this thesis.
To test that the \gls*{nm} method works as intended a number of trial optimisation functions were tested, see~\cref{tab:testfuncs}.
This was achieved by selecting an initial simplex, and the method allowed to iterate until it converged.
The results of this are shown in~\cref{fig:nmtest}.


\begin{table}[!htbp]
    \begin{tabular}{|c|c|c|}
    \hline
        Name         & Formula                                                                & Global Minumum                \\ \hline
        Sphere       & $x^2+y^2$                                                              & $f(0,0)=0.$                   \\ \hline
        Rosenbrock   & $(a-x)^2+b(y-x^2)^2$                                                   & $f(1,1)=0.$                   \\ \hline
        Ackely       & $ -20\exp\left[-0.2\sqrt{0.5\left(x^{2}+y^{2}\right)}\right] - $       & $f(0,0)=0.$                   \\
                     & $\exp\left[0.5\left(\cos 2\pi x + \cos 2\pi y \right)\right] + e + 20$ &                               \\ \hline
        Himmelblau's & $(x^2+y-11)^2+(x+y^2-7)^2$                                             & $f(3,2)=0., $                 \\
                     &                                                                        & $f(-2.805118,3.131312)=0.,$   \\
                     &                                                                        & $f(-3.779310,-3.283186)=0.,$  \\  
                     &                                                                        & $f(3.584428,-1.848126)=0.$    \\ \hline
    \end{tabular}
    \caption{Table of standard test functions for numerical optimisation.}
    \label{tab:testfuncs}
\end{table}


\begin{figure}[!htbp]
    \centering
    \includegraphics[width=0.75\textwidth]{all-nm-tests.pdf}
    \caption{Contour plots of test functions with Nelder-Mead simplexes over plotted. Top left is the Ackely function, top right is the sphere function, bottom left is the Himmelblau's function, and the bottom right is the Rosenbrock function. Blue simplex is the initial simplex, and the large black dots represent the Global minima.}
    \label{fig:nmtest}
\end{figure}

Some of these functions (Sphere, and Rosenbrock's) can also be extended to arbitrary dimensions. These functions were used to check that the NM method works as intended in these higher dimensions where the NM method will primarily be used in this thesis.

To ensure that the~\gls*{nm} method can be used to find the unknown concentrations of the autoflurophores, we test the method with a two different toy models.
The first model is a 2D model, with two different fluorophores evenly distributed over all 5 layers of our skin model. 
The two different fluorophores are NADH, and a fictitious fluorophore that has similar properties to FAD and tryptophan, such that the excitation spectrum is that of FAD and the emission spectrum is that of tryptophan.
The concentration in these layers is such that the bulk optical properties are not affected: NADH has a concentration of $1.0~\mu M$, and the fictitious fluorophore has a concentration of $2.5~\mu M$.
To generate a spectrum to which the \gls*{nm} method can compare to, the \gls*{mcrt} code is run with the above configuration of fluorophores.
This generated~\cref{fig:toymodelspectra}.


\begin{figure}[!htbp]
  \centering
  \includegraphics[width=0.6\textwidth]{target-toy.pdf}
  \caption{Example of toy model spectrum for testing NM method. The two peaks correspond to the fictitious fluorophore, and NADH.}
  \label{fig:toymodelspectra}
\end{figure}

The fitness function chosen to check whether the NM method is converging to the generated spectrum is as:

\begin{equation}
fitness = \sum\limits_{i=1}^{n}(x_i-m_i)^2
\end{equation}

Where $x_i$ is a data point at a wavelength $\lambda_i$ produced by the MCRT, and $m_i$ is a data point in the model at a wavelength $\lambda_i$.

As the \gls*{nm} can get stuck in local minima, we run the method for several different initial simplices to ensure that this does not occur.
As the \gls*{mcrt} code is called multiple times per simplex iteration, and the fluorophore concentration is low meaning that many packets need to be run to achieve a good signal to noise ratio.
These two conditions result in a computational load that is infeasible to run. 
Therefore the~\gls*{mcrt} algorithm has to be computationally efficient in order to arrive at an answer within a reasonable time.
To this end the 3D skin model is shrunk to a 1D model in the z direction so that the optical integration routine can efficiently move the packet through the simulated medium.
The optical properties of the incident wavelength are also stored so that when a new packet is started the optical properties can easily be adjusted without need for any calculation.
Finally a filter is employed on the output fluorescence spectrum to smooth the noise out.
The filter used is a Savitzky-Golay filter.
This filter fits multiple low-degree polynomials to subsets of the data, thus smoothing the data~\cite{press1990savitzky}. 
\Cref{fig:sgfilter} shows the use of the Savitzky-Golay filter on sample output data from the \gls*{mcrt} simulation.

\begin{figure}[!htbp]
  \centering
  \includegraphics[width=0.75\textwidth]{sgfilter-prrof.pdf}
  \caption{Illustration of how the Savitzky-Golay filter works on noisy data and recovers the roughly the same signal on the same data set with less noise. Left image shows the raw signal for a simulation run with less photons and more photons. Right image shows the data set after the Savitzky-Golay filter is applied. A window size of 101, and polynomial of order 2 were used as the filter settings.}
  \label{fig:sgfilter}
\end{figure}


Using the above set-up with the Savitzky-Golay filter on the output spectrum, allowed the \gls*{nm} method to efficiently run many model of various different concentrations and ``find'' a set of concentration that resulted in a close match with the target spectrum.
However, as the detected fluorescence spectra are normalised to their peak values we cannot use this method to determine the original concentration, but rather the ratio between the two concentration.
\Cref{fig:spaceplot2D} shows the search space and the spread of the concentration values calculated by the \gls*{nm} method compared to the original target concentration.
The spread of the concentrations calculated by the \gls*{nm} method follow a linear relationship as would be expected.
Therefore, a line of best fit was fitted to the concentrations.
This yielded a line ($y=m\ x$) with $m=2.49\pm0.05$.
The expected relation between the concentration is where $m =2.5$ therefore, the \gls*{nm} method can be used to determine the relative difference in concentrations within one standard deviation.


\begin{figure}[!htpb]
  \centering
  \includegraphics[width=0.65\textwidth]{2d-space-plot-thesis.pdf}
  \caption{Figure shows the search space for the 2D toy problem outlined above. A line of best fit is fitted to the concentrations found by the \gls*{nm} method. Note also the valley of good fit where the line of best fit lies. The search space is also fairly smooth.}
  \label{fig:spaceplot2D}
\end{figure}


The \gls*{nm} method was also tested on a toy model for $n=3$.
Here the fluorophores used were: NADH, FAD, and a fictitious fluorophore with the absorption properties of NADH and the emission properties of Tyrosine.
The fluorophores had concentrations of $1.05\mu M$ $525\mu M$, and $125\mu M$ respectively.
The set-up is the same for the above $n=2$ case, with the same filter and computational speed ups used.
\Cref{fig:3dtoymodel} shows various concentrations as calculated by the \gls*{nm} method compared to the original concentration.
Again a line of best fit was fitted to the calculated points \footnote{The line of best fit was calculated using the singular value decomposition (SVD) method of solving the least squares.}.
As there is no closed form solution for the equation of a line in $n>2$ dimensions, the equation of a line in 3D can be represented by~\cref{eqn:3dlineeqn}:

\begin{equation}
\overrightarrow{r}=\overrightarrow{r_0}+t\overrightarrow{v}
\label{eqn:3dlineeqn}
\end{equation}

Where $\overrightarrow{r}$ and $\overrightarrow{r_0}$ are position vectors for $P$ and $P_0$ respectively, $\overrightarrow{v}$ is a vector parallel to the line we are examining, and t is some real number.
This equation also holds for an arbitrary number of dimensions.
The line of best fit yielded $\overrightarrow{v} =\left[0.0013, 0.9858, 0.1682\right]$.
Where as the ``real'' $\overrightarrow{v}$ is equal to $\left[0.0019, 0.9728, 0.2316\right]$.
This again shows that the NM method can be used to find the relative concentrations of the fluorophores in the skin, even in dimensions higher than 2.

\begin{figure}[!htpb]
  \centering
  \includegraphics[width=0.75\textwidth]{3d-nm-example.pdf}
  \caption{Figure shows the line of best fit for the case where $n=3$. Figure also shows the simplices path over their whole lifetime, from initial guess to final simplex.}
  \label{fig:3dtoymodel}
\end{figure}

\FloatBarrier
\section{Results}

\subsection*{Experimental Work}

Experimental work discussed in this section was carried out by my collaborators S. Smirni \textit{et al.} at the University of Dundee and Ninewells Hospital.

S. Smirni \textit{et al.} took autofluorescence spectra from the volar forearms of volunteers using the LAKK-M multi-functional laser non-invasive diagnostic system (MLNDS)\@.
The LAKK-M system combines laser-Doppler flowmetry, tissue reflectance oximetry, pulse oximetry, and laser fluorescent diagnostics in one machine.
The LAKK-M system delivers and collects light via the same optical fiber package.
The fiber package has seven optical fibers, one for each probing wavelength (365~nm, 430~nm, 532~nm, 635~nm), two detectors, and a spectrometer.
The fibers radii are 0.4~mm, and the separation between the fibers is around 1~mm.
S. Smirni \textit{et al.} took autofluorescence spectra from volunteers as they underwent a \gls*{porh} test.
A \gls*{porh} test is used to investigate and asses microvascular function.
It achieves this by usually inflating a cuff on an arm of a patient for a period of time and then letting it deflate.
Whilst this is happening laser Doppler flowmetry is used to asses the flow of blood before inflation, during inflation and after inflation.
How the micro-vascular system responds to this test can be indicative of various \gls*{cvds}.
\Cref{fig:porh} shows an example of the data collected during a \gls*{porh} test.


\begin{figure}[!htpb]
  \centering
  \includegraphics[width=0.5\textwidth]{porh-example.pdf}
  \caption{Example of a PORH test whilst measuring autofluorescence and perfusion. The perfusion of blood in the skin decrease once the cuff is inflated, and then rapidly reaches a maximum once the cuff is removed. The inverse of this can be seen in the autofluorescent response of NADH in the tissue.}
  \label{fig:porh}
\end{figure}

S. Smirni collected around 27 spectra at the rate of 1 per minute over a period of 27 minutes as the \gls*{porh} test was undertaken, 11 for a baseline before inflation of the cuff, 5 during occlusion and 11 post occlusion.
Spectra were taken at each of the possible wavelengths (635~nm, 532~nm, and 365~nm) that the LAKK-M provides, yielding a total of 81 autofluorescence spectra per volunteer.
S. Smirni also collected laser Doppler flowmetry data during this period, however as this work only concerns the modelling of autofluorescence, we restrict the analysis to that of the UV (365~nm) autofluorescence spectra.
\Cref{fig:slavodataexample} shows an example of a raw spectrum taken during the baseline portion of the experiment.

\begin{figure}[!htpb]
  \centering
  \includegraphics[width=0.5\textwidth]{raw-spectra.pdf}
  \caption{Figure shows an example of a raw spectrum taken by S. Smirni \textit{et al.}. Figure illustrates how the spectrum is red-shifted, along with some of the artifacts, backscatter, and autofluorescence peaks. There is also an third peak in the red end of the spectrum. The cause of this is unknown, but most likely due to an unidentified fluorophore.}
  \label{fig:slavodataexample}
\end{figure}

\subsection*{Effect of Tissue Optics on Fluorescent Signal}

As mentioned in the introduction to this chapter information about how tissue optics affects the fluorescent signal, which fluorophores contribute to the signal, the variability of the signal in different test sites, and location of the fluorescent signal are not well elucidated. 
Therefore, before running the \gls*{nm} method on the experimental data, the fluence of the input fluorescence light is analysed alongside the location as a function of depth of the fluorescent light.
We also present results on how the excitation wavelength affects the signal.

\Cref{fig:uvpen} shows the fluence as a function of depth for the incident UV light.
The figure shows that most of the incident light is contained within the top three layers, with little getting to the Reticular dermis, with none reaching the Hypodermis.
The fluence drops to 50\% of its peak value 0.026~cm into the tissue which equates to inside epidermis, and 10\% of its peak value by 0.058~cm into the tissue which again is in the epidermis.

\begin{figure}[!htpb]
    \centering
    \includegraphics[width=0.75\textwidth]{uv-fluence.pdf}
    \caption{Penetration of UV radiation as a function of depth.}
    \label{fig:uvpen}
\end{figure}

\Cref{fig:fadnadhboth} shows the fluence of detected fluorescent light (see~\cref{app:lightdect} for discussion of how this is tracked.).
The figure shows the fluence of light for each fluorophore that is detected.
The figure shows that the fluence is highest in the Papillary dermis and a number of peaks at the boundaries of the layers.
First the refractive indices of the layers are different, this can lead to light getting ``trapped'' in a layer as it maybe reflected off the layer boundary, thus leading to increased fluence.
Second, fluorescent light is emitted isotropically which means that fluorescent light emitted in the upper layers of the skin, may be emitted in the direction of the Papillary dermis, that and light emitted from below the papillary dermis has to travel though the papillary dermis in order to be detected.
Finally the optical properties also have an effect on the detected light fluence.
The dermal layers have a minimum in their absorption spectra at the range of wavelengths that NADH and FAD emit at, thus light is more likely to ``survive'' though the dermal layers than the other layers such as the epidermis.
Finally the geometry of the layers most likely has an effect as well.
The stratum corenum's and the epidermis's thickness is small in comparison to that of the dermal layers, thus there will be less fluorescent light emitted from these layers when the concentration are the same.

\begin{figure}[!htpb]
    \centering
    \includegraphics[width=0.75\textwidth]{detect-flu-zoomin.pdf}
    \caption{Detected fluence for FAD, NADH, and elastin fluorescence. Inset show zoom in of top layers of the skin, note the inset is a linear scale.}
    \label{fig:fadnadhboth}
\end{figure}

\Cref{fig:floc} shows the location of where the fluorescent light is emitted from.
For both NADH, and FAD there peaks depth is just inside the epidermis.
However, the vast majority of emitted Fluorescent light originates in the papillary dermis.
This occurs as enough light gets to this layer, which allows more light to under go fluorescence. 


\begin{figure}[!htpb]
    \centering
    \includegraphics[width=0.75\textwidth]{fluro-loc.pdf}
    \caption{Amount of packets escaping as a function of depth for FAD and NADH fluorescence.}
    \label{fig:floc}
\end{figure}

Finally, we analyse the effect of excitation wavelength has on the detected signal of NADH and FAD\@.
To achieve this we run several models in order to create an \gls*{eem}.
A \gls*{eem} is a way of investigating and displaying the effect of excitation wavelength has on the fluorescent output of a sample.
Using the same setup as above, we vary the fluorophore concentration in each layer, such that only one layer at a time has any fluorophores in it.
We then excite the model over a range of wavelengths.
For NADH we use wavelength in the range of 250 to 400~nm, and for FAD we use 250 to 500~nm.
\Cref{fig:nadheem,fig:fadeem} show the \gls*{eem} for NADH and FAD\@.
The \gls*{eem} for NADH shows that the strongest signal comes from the papillary dermis when compared to other layers.
The \gls*{eem} also shows a maximum of emission for excitation wavelengths corresponding to around 260~nm and 320--380~nm.
For FAD the layer that yields the most fluorescence is the reticular dermis.
The \gls*{eem} shows that for excitation wavelengths of around the range 450--500~nm gives the most fluorescence.

For the case of NADH the optimal excitation wavelengths coincide with maximums in the absorption spectrum of NADH (see~\cref{fig:epsfluro}).
However, for FAD the maximum in fluorescence does not coincide with the largest maxima in its absorption spectrum (see~\cref{fig:epsfluro}), but rather it falls within a smaller maxima.
This is because that light at the maximal absorption peak ($\sim$300~nm) is highly absorbed by tissue, especially in layers such as the epidermis.
Therefore, light at a longer wavelength is more likely to escape the tissue due to the tissue optical properties.

\begin{figure}[!htpb]
    \centering
    \includegraphics[width=0.75\textwidth]{show-eps-and-fluro-out.pdf}
    \caption{NADH and FAD absorption and emission spectra.}
    \label{fig:epsfluro}
\end{figure}

\begin{figure}[!htpb]
    \centering
    \includegraphics[width=0.6\textwidth]{NADH-eem.pdf}
    \caption{Excitation-emission matrix for NADH. Figure shows that the fluorescent signal from NADH is strongest in the papillary dermis.}
    \label{fig:nadheem}
\end{figure}

\begin{figure}[!htpb]
    \centering
    \includegraphics[width=0.6\textwidth]{FAD-eem.pdf}
    \caption{Excitation-emission matrix for FAD. Figure shows that the fluorescent signal from FAD is strongest in the reticular dermis.}
    \label{fig:fadeem}
\end{figure}

\subsection*{Using the NM method}

We model the experimental setup by S. Smirni \textit{et al.} as described in a previous section.
The computational speed ups described in the validation sections are also used.
The 5 layers skin model is setup with NADH, FAD and collagen distributed in different layers.
NADH and FAD are initially in all layers bar the hypodermis, and collagen is contained in just the papillary and reticular dermis\footnote{We omit any fluorophores form the hypodermis as they are not expected to contribute to the signal at all, or by much. This also simplifies the problem from n=13 to n=10.}.
Each layer has its own concentration of the fluorophores giving $n$ for the \gls*{nm} method of 10.
An initial guess of the concentration of the fluorophores is made, and the \gls*{nm} method is allowed to run until it stagnates.
The target spectrum for the \gls*{nm} is taken from S. Smirni experimental data.
Before the target spectrum could be used it was first ``cleaned up''.
This ``cleaning'' process was required as the spectrum was red shifted by 20~nm.
The cause of this redshift is unknown.
As a backscatter peak is included within the spectrum, the spectrum can be moved to it correct position.
The next step in the ``cleaning'' process is to filter out spectra that are ``defective''.
Several of the autofluorescence spectra have artificial peaks caused by various experimental errors or machine faults.
The final stage of the ``cleaning'' process, is to smooth the data out and remove the backscatter peak. 
Now the spectra can be used as a target spectrum for the \gls*{nm} method.

\Cref{fig:NMupdate} shows the output of the \gls*{nm} method during a run with the above parameters.
The figure shows that as the simplex iterates, the average fitness improves, the average number of calls to the \gls*{mcrt} algorithm tends to $\sim 1.4$.
This suggests that the \gls*{nm} keeps the reflected point more often than not.
The size of the simplex fluctuates as the simplex tries to find the global minima.
This happens as the simplex moves over the search space it sometimes needs to grow in size, then when it finds a minima it shrinks around it.
The size of the simplex will then grow when the algorithm is restarted.

\begin{figure}[!htpb]
  \centering
  \includegraphics[width=0.6\textwidth]{NM-update.pdf}
  \caption{Figure shows the diagnostic information output by the \gls*{nm} method. Top left) shows the size of the simplex as it evolves. Top right) shows the best point in the simplex as it evolves. Bottom left) shows the average fitness of the simplex. Bottom right) shows the number of calls to the MCRT code per iteration of the simplex. The more call to the MCRT the simplex requires the higher the runtime.}
  \label{fig:NMupdate}
\end{figure}

\Cref{fig:bestNMresult} shows the output of the \gls*{nm}.
The ``best'' fitness achieved by the algorithm was $\sim3$, which is a fairly poor fit when compared to the validation examples above.
The figure also shows the contribution of the individual fluorophores to the overall autofluorescence signal.
As expected the main contribution is from NADH, with a little contribution by FAD\@.
However, collagen contributes to the main peak and is multi-peaked, with peaks at $\sim$380~nm and $\sim$ 465~nm.
This is unusual as the peak of emission for collagen used in this work is around 395~nm, therefore it was not expected to contribute to the signal around the NADH peak.
The reason that the collagen signal is multi-peaked is due to the tissue optics.
There is a large peak at around 400--410~nm in the absorption spectra in the dermal layers.
This causes a dip in the emission of collagen, giving it the multi-peaked profile.

\begin{figure}[!htpb]
  \centering
  \includegraphics[width=0.75\textwidth]{nm-bestfit.pdf}
  \caption{Figure shows the best result from the \gls*{nm} operating on the experimental data from S. Smirni \textit{et al.}. The spectrum has a fitness of $\sim 3$. The figure also shows the makeup of the final spectrum by the individual fluorophores.}
  \label{fig:bestNMresult}
\end{figure}

\section{Discussion}

There are several probable reasons why the spectrum generated by the \gls*{mcrt} and \gls*{nm} does not match the experimental data exactly.
A first possible reason for the mismatch, is that the optical properties of the fluorophores are not 100\% correct.
Optical properties of the fluorophores used in the work are not readily available.
There are various sources that have measured them, but they are usually confined to several wavelengths of the authors interest and not a full spectrum.
The extinction coefficients for the fluorophores are measured in various buffer solutions, and not taken from \textit{in-vivo} tissue.
This could give rise to a discrepancy between the computed results and the experimental results.
The emission spectra for the fluorophores could also be incorrect.
Eission spectra used in this work are also measured in various buffer solutions, which again may be different from what they are \textit{in-vivo}.
Finally the emission spectra are measured for a single excitation wavelength.
As the spectra were not readily available for the excitation wavelength used in this work, the emission spectra were taken from the nearest possible excitation wavelength found in the literature. This again may introduce a difference from the experimental work.
However, what optical properties that have been published are pretty consistent from author to author therefore, this is an unlikely problem.

The mismatch in the target spectrum and the calculated spectrum could be due to missing fluorophores. 
The model presented here only included NADH, FAD, and collagen which are thought to be the three main contributers to the autofluorescence signal.
However, the inclusion of riboflavin (peak at 529~nm), $\beta$-carotene (395~nm), elastin(425~nm), and different types of collagen (420, 370, 395~nm) may be able to correct this.

A final possible reason for the mismatch is that the LAKK-M device creators seem to have mis-calibrated their machines for finding the peaks of various fluorophores.
\Cref{tab:flurocompvalues} shows a comparison between the literature and the values used by the LAKK-M team.

\begin{table}[!htbp]
  \centering

  \begin{tabular}{l|cc}

  \hline
  Fluorophore & LAKK-M peak/nm & Literature peak/nm \\
  \hline
   FAD  & 550& 525\\
   NADH  & 490& 465\\
   Elastin  & 450& 425\\
   Collagen  & 420 & 370(IV), 395(I,V), 600(VII)\\

  \hline

  \end{tabular}
  \caption{Comparison of LAKK-M emission peaks and that found in the literature. AS there are different types of collagen multiple types were included in the literature comparison. LAKK-M data from~\cite{palmer2016changes,dunaev2015individual,smirnova2012collagen,palmer2016development}. Literature values taken from~\cite{dacosta2003molecular,pan2015detection,islam2013ph,patterson2000separation}.}
  \label{tab:flurocompvalues}

\end{table}

For whatever reason, they seem to be consistently off by around 25 to 30~nm.
From this chapters work on how tissue optics affects the fluorescent signal, tissue optics cannot account for the discrepancy.
One possible cause for this is that the UV light they use to probe autofluorescence is an LED\@.
LED light sources are not monochromatic, but rather have a range of wavelengths they emit, on the order of 25~nm FWHM\@.
There they could be exciting with a different range of wavelengths, which could possibly contribute towards the difference in what they measure and the literature values.

\section{Conclusion}

In conclusion, this section has introduced a five layer skin model, that has the ability to vary the various constitute parts to tailor model for a generic or specific patient.
We have also shown in this chapter a method of coupling an optimisation technique to \gls*{mcrt}.
This was done in order to investigate the concentrations of naturally occurring fluorophores in the skin.
Differing levels of the fluorophores can indicate various diseases such as diabetes and \gls*{cvds}.
We validated this technique against a toy model in various dimensions, with good agreement.
Before experimental data was anyalsed, the effect of tissue optics on the autofluorescent signal was tested.
We then used our numerical model on experimental data. We found good, but not excellent agreement with the data.
A discussion of why our model does not fit the experimental data, with several possible reasons presented.
Future work would involve taking known ``good'' data that would be worthwhile to validate against.
Finally, if the model can be validated against this data the model could be used in future to explore the relationship between various diseases and autofluorescence.


%No paper or aim for project. Need contact with Dundee/Ninewells to proceed.
